### PR TITLE
import PropTypes from ‘prop-types’ and ViewPropTypes instead of View.propTypes

### PR DESCRIPTION
**Pull request by bfanger**
Fixed errors for react-native 0.44.3 and up:

`View.propTypes has been deprecated and will be removed in a future version of ReactNative. Use ViewPropTypes instead.`

`Warning: PropTypes has been moved to a separate package. Accessing React.PropTypes is no longer supported and will be removed completely in React 16. Use the prop-types package on npm instead.`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
   },
   "peerDependencies": {
     "react": ">=15.4.0",
-    "react-native": ">=0.40"
+    "react-native": ">=0.44.3",
+    "prop-types": "*"
```
Comment (henrikra) [inline package.json]: Do not EVER user `*` as dependency version :O Use for example `^15.5.10`

File: Annotation.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
-import React, { PropTypes } from 'react';
+import PropTypes from 'prop-types'
```
Comment (henrikra) [inline Annotation.js]: Missing semicolon

File: index.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 'use strict';
 
-import React, { Component, PropTypes } from 'react';
+import PropTypes from 'prop-types'
```
Comment (henrikra) [inline index.js]: Missing semi colon here too

**Comment by kurt-o-sys** [comment]
+1  Absolutely necessary. mapbox-gl doesn't work anymore in 0.46.

**Comment by bfanger** [comment]
Made the changes requested by @henrikra

**Comment by kurt-o-sys** [comment]
Right... waiting for it to merge in master?

**Comment by henrikra** [comment]
@kurt-o-sys I just tried fresh install with RN 0.46 and latest version of this map library and they worked together! It just gives warning about this ViewTypes thing :) But yeah this should be merged anyways.

**Comment by kurt-o-sys** [comment]
weird... let me try again. RN 0.46.4? and which version of react? 16.0.0-alpha.13? (or -alpha.12?)

**Comment by henrikra** [comment]
React: `16.0.0-alpha.12`
React Native: `0.46.4`

What error you are getting and when?

**Comment by kurt-o-sys** [comment]
oh... that might be it. I found react `16.0.0-alpha.13` to give issues...  `alpha.12` seems to be ok. Thx!

**Comment by henrikra** [comment]
@zugaldia I just tried @bfanger 's branch and this PR indeed gets rid of those warnings in the console! So it is ready to be merged.

**Comment by zugaldia** [comment]
Hey, I just wanted to give y'all an update - good news, soon the Mapbox team is gonna grow to have dedicated resources for this project 🎉 . This means that we'll be able to move much faster fixing and merging PRs, please bear with us while we internally cross the t's and dot the i's.

**Comment by henrikra** [comment]
@zugaldia Is this going to happen this week or what? :D

**Comment by zugaldia** [comment]
@henrikra https://github.com/mapbox/react-native-mapbox-gl/issues/590#issuecomment-322901528 ;-)

**Comment by nitaliano** [comment]
@bfanger :+1: this LGTM!

**Comment by jjzazuet** [comment]
Hi. Same here. I'm playing code janitor with the rest of the components my app is using. Please update NPM. Thanks for your time and help!
